source/docs/stats.md

```diff
@@ -184,7 +184,14 @@ showing the memory layout for a few scenarios of constructing and joining symbol
 
 There are several ways to create hot-path contention looking up stats by name,
 and there is no bulletproof way to prevent it from occurring.
- * The [stats macros](https://github.com/envoyproxy/envoy/blob/master/include/envoy/stats/stats_macros.h) may be used in a data structure which is constructed in response to requests.
+ * The [stats macros](https://github.com/envoyproxy/envoy/blob/master/include/envoy/stats/stats_macros.h) may be used in a data structure which is constructed in response to requests. In this
+   scenario, consider factoring out the symbolization phase using MAKE_STAT_NAMES_STRUCT
+   in a factory or context during startup, and using MAKE_STATS_STRUCT in the hot-path,
```

Comment:
It's not just about requests, right? I think a bunch of your motivation was actually in response to control plane updates on the main thread? Do you want to comment about that also?

/wait

Reply:
good point; done.